pkg/util/containers/cri/util.go

```diff
@@ -104,6 +108,53 @@ func GetUtil() (*CRIUtil, error) {
 	return globalCRIUtil, nil
 }
 
+// sts begin
+
+// ContainerStateMap is used to map cri specific state to own internal state
+var ContainerStateMap = map[pb.ContainerState]string{
+	pb.ContainerState_CONTAINER_CREATED: containers.ContainerCreatedState,
+	pb.ContainerState_CONTAINER_RUNNING: containers.ContainerRunningState,
+	pb.ContainerState_CONTAINER_EXITED:  containers.ContainerExitedState,
+	pb.ContainerState_CONTAINER_UNKNOWN: containers.ContainerUnknownState,
+}
+
+func (c *CRIUtil) GetContainers() ([]*spec.Container, error) {
+	containerStats, err := c.ListContainerStats()
+	if err != nil {
+		return nil, err
+	}
+	uContainers := make([]*spec.Container, 0, len(containerStats))
+	for cid := range containerStats {
+		cstatus, err := c.GetContainerStatus(cid)
+		if err != nil {
+			log.Debugf("Could not get status of container '%s'", cid)
+			continue
+		}
+		mounts := make([]specs.Mount, 0, len(cstatus.Mounts))
+		for _, cmount := range cstatus.Mounts {
+			mountPoint := specs.Mount{
+				Source:      cmount.HostPath,
+				Destination: cmount.ContainerPath,
+			}
+			mounts = append(mounts, mountPoint)
+		}
+		container := &spec.Container{
+			Runtime: "CRI",
+			Name:    cstatus.Metadata.Name,
+			ID:      cid,
+			Image:   cstatus.Image.Image,
+			Mounts:  mounts,
+		}
+		if state, ok := ContainerStateMap[cstatus.State]; ok {
```

Comment:
Add a log here if the `cstatus.State` is not found in the `ContainerStateMap`, it probably means we're missing some values in our map

Reply:
Also warning :smile: